device/cuda/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cu

```diff
@@ -0,0 +1,487 @@
+                                           stream>>>(
+                device::sort_updated_tracks_payload{
+                    .rel_shared_view = rel_shared_buffer,
+                    .pvals_view = pvals_buffer,
+                    .update_res = update_res_device.get(),
+                    .updated_tracks_view = updated_tracks_buffer,
+                });
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+
+            kernels::fill_inverted_ids<<<nBlocks, nThreads, 0, stream>>>(
+                device::fill_inverted_ids_payload{
+                    .sorted_ids_view = sorted_ids_buffer,
+                    .update_res = update_res_device.get(),
+                    .inverted_ids_view = inverted_ids_buffer,
+                });
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+
+            kernels::block_inclusive_scan<<<nBlocks, nThreads, 32 * sizeof(int),
+                                            stream>>>(
+                device::block_inclusive_scan_payload{
+                    .sorted_ids_view = sorted_ids_buffer,
+                    .update_res = update_res_device.get(),
+                    .is_updated_view = is_updated_buffer,
+                    .block_offsets_view = block_offsets_buffer,
+                    .prefix_sums_view = prefix_sums_buffer});
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+
+            kernels::add_block_offset<<<nBlocks, nThreads, 0, stream>>>(
+                device::add_block_offset_payload{
+                    .update_res = update_res_device.get(),
+                    .block_offsets_view = block_offsets_buffer,
+                    .prefix_sums_view = prefix_sums_buffer});
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+            
+            kernels::rearrange_tracks<<<nBlocks, nThreads, 0, stream>>>(
+                device::rearrange_tracks_payload{
+                    .sorted_ids_view = sorted_ids_buffer,
+                    .inverted_ids_view = inverted_ids_buffer,
+                    .rel_shared_view = rel_shared_buffer,
+                    .pvals_view = pvals_buffer,
+                    .update_res = update_res_device.get(),
+                    .updated_tracks_view = updated_tracks_buffer,
+                    .is_updated_view = is_updated_buffer,
+                    .prefix_sums_view = prefix_sums_buffer,
+                    .temp_sorted_ids_view = temp_sorted_ids_buffer,
+                });
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+
+            kernels::gather_tracks<<<nBlocks, nThreads, 0, stream>>>(
+                device::gather_tracks_payload{
+                    .update_res = update_res_device.get(),
+                    .temp_sorted_ids_view = temp_sorted_ids_buffer,
+                    .sorted_ids_view = sorted_ids_buffer,
+                    .is_updated_view = is_updated_buffer});
+            TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
```

Comment:
:thinking: These kernels could be merged into a single one, no? Many of them are launched in the exact same way. Some of them are a bit different, both shared memory and thread-count-wise, but it would still seem possible to do this in one go, no?

Reply:
Not in one go - mostly because of synchronization issue between blocks. 
BTW, we don't have to worry about the launch overhead now - I tied them inside the cudagraph loop